internal/api/mfa.go

```diff
@@ -90,10 +90,27 @@ func (a *API) enrollPhoneFactor(w http.ResponseWriter, r *http.Request, params *
 	if err := models.DeleteExpiredFactors(db, config.MFA.FactorExpiryDuration); err != nil {
 		return err
 	}
-
-	for _, factor := range factors {
-		if factor.IsVerified() {
-			numVerifiedFactors += 1
+	for _, factor := range user.Factors {
+		switch {
+		case factor.FriendlyName == params.FriendlyName:
+			return unprocessableEntityError(
+				ErrorCodeMFAFactorNameConflict,
+				fmt.Sprintf("A factor with the friendly name %q for this user already exists", factor.FriendlyName),
+			)
+
+		case factor.IsVerified() && factor.IsPhoneFactor():
+			return unprocessableEntityError(
+				ErrorCodeVerifiedPhoneFactorExists,
+				"A verified phone factor already exists, unenroll the existing factor to continue",
+			)
+
+		case factor.IsVerified():
+			numVerifiedFactors++
+
+		case factor.IsUnverified() && factor.IsPhoneFactor():
+			if err := db.Destroy(factor); err != nil {
+				return internalServerError("Database error deleting factor").WithInternalError(err)
+			}
```

Comment:
hmm so we just destroy all unverified phone factors even if they are using different phone numbers? i thought we should just be destroying only the factor that is using the same phone number as the one currently being enrolled?

Reply:
>hmm so we just destroy all unverified phone factors even if they are using different phone numbers? 
Initial consideration was that they should only be using a single phone number since we ultimately only allow for one verified factor. 

If they are multiple phone factors enrolled the thought process was that the long tail of users would only care about completing verification on the latest factor.

>i thought we should just be destroying only the factor that is using the same phone number as the one currently being enrolled?
I agree that this is more precise and leaves less room for edge cases though. Let's go with this